lang/cs/Org.Apache.REEF.IO/PartitionedData/IInputPartition.cs

```diff
@@ -24,23 +25,24 @@ namespace Org.Apache.REEF.IO.PartitionedData
     /// </summary>
     /// <typeparam name="T">Generic Type representing data pointer.
     /// For example, for data in local file it can be file pointer </typeparam>
-    public interface IInputPartition<T>
+    [Unstable("API contract may change.")]
+    public interface IInputPartition<T> 
     {
         /// <summary>
         /// The id of the partition.
         /// </summary>
         string Id { get; }
 
         /// <summary>
-        /// Caches the data locally, cached location is based on the implementation.
+        /// Caches the data based on the method parameter. Returns the actual cached level.
         /// </summary>
-        [Unstable("0.14", "Contract may change.")]
-        void Cache();
+        [Unstable("0.15", "Contract may change.")]
+        CacheLevel Cache(CacheLevel level);
 
         /// <summary>
         /// Gives a pointer to the underlying partition.
         /// </summary>
         /// <returns>The pointer to the underlying partition</returns>
-        T GetPartitionHandle();
+        IEnumerable<T> GetPartitionHandle();
```

Comment:
Yes. We need this because we would ideally like to be able to map a Stream to an object with the new `ISerializer` interface. The `ISerializer` does not have a natural way to map multiple files to a single `T`. This allows more flexibility - the user can always merge streams and return a single `T` if that is what they would like in their implementations.